packages/portfolio-contract/src/type-guards.ts

```diff
@@ -68,72 +87,117 @@ export const makeProposalShapes = (usdcBrand: Brand<'nat'>) => {
+    destinationEVMChain: M.or(...keys(AxelarChains)),
+  },
+) as TypedPattern<EVMOfferArgs>;
 
-// Use Axelar chain identifier instead of CAP-10 ID for cross-chain messaging
-// Axelar docs: https://docs.axelar.dev/dev/reference/mainnet-chain-names
-// Chain names: https://axelarscan.io/resources/chains
-const optEvmChainShape = M.splitRecord({}, { evmChain: ChainShape });
+export type OfferArgsFor = {
+  openPortfolio: EVMOfferArgs;
+  rebalance: EVMOfferArgs;
+};
 
 export const OfferArgsShapeFor = {
-  openPortfolio: optEvmChainShape as TypedPattern<
+  openPortfolio: EVMOfferArgsShape as TypedPattern<
     OfferArgsFor['openPortfolio']
   >,
-  rebalance: optEvmChainShape as TypedPattern<OfferArgsFor['rebalance']>,
+  rebalance: EVMOfferArgsShape as TypedPattern<OfferArgsFor['rebalance']>,
 };
 harden(OfferArgsShapeFor);
 
-export type GMPArgs = {
-  destinationAddress: string;
-  destinationEVMChain: SupportedEVMChains;
-} & (
-  | { type: 3 }
-  | {
-      type: 1 | 2;
-      gasAmount: number;
-      contractInvocationData: Array<ContractCall>;
-    }
-);
-
-export const ContractCallShape = harden({
-  target: M.string(),
-  functionSignature: M.string(),
-  args: M.array(),
-});
-
-export const GMPArgsShape: TypedPattern<GMPArgs> = M.and(
-  M.splitRecord({ destinationAddress: M.string() }),
-  M.or(
-    M.splitRecord({ type: 1 }),
-    M.splitRecord({
-      type: M.or(1, 2),
-      gasAmount: M.number(),
-      contractInvocationData: M.arrayOf(ContractCallShape),
-    }),
-  ),
-);
-
 export type EVMContractAddresses = {
-  aavePool: string;
-  compound: string;
-  factory: string;
+  aavePool: `0x${string}`;
+  compound: `0x${string}`;
+  factory: `0x${string}`;
+  usdc: `0x${string}`;
 };
 
 export const EVMContractAddressesShape: TypedPattern<EVMContractAddresses> =
   M.splitRecord({
     aavePool: M.string(),
     compound: M.string(),
     factory: M.string(),
+    usdc: M.string(),
   });
 
+const AxelarChainInfoPattern = M.splitRecord({
+  caip: M.string(),
+  axelarId: M.string(),
+});
+
+export const AxelarChainsMapShape: TypedPattern<AxelarChainsMap> =
+  M.splitRecord(
+    fromEntries(
+      keys(AxelarChains).map(chain => [chain, AxelarChainInfoPattern]),
+    ) as Record<AxelarChain, typeof AxelarChainInfoPattern>,
+  );
+
+export type PortfolioBootstrapContext = {
+  axelarChainsMap: AxelarChainsMap;
+  chainHubTools: {
+    getDenom: (brand: Brand) => Denom | undefined;
+  };
+  contractAddresses: EVMContractAddresses;
+  zoeTools: GuestInterface<ZoeTools>;
+  makePortfolioKit: () => GuestInterface<PortfolioKit>;
+  inertSubscriber: GuestInterface<ResolvedPublicTopic<never>['subscriber']>;
+};
+
+export type PortfolioInstanceContext = {
+  axelarChainsMap: AxelarChainsMap;
+  chainHubTools: {
+    getDenom: (brand: Brand) => Denom | undefined;
+  };
+  contractAddresses: EVMContractAddresses;
+  inertSubscriber: GuestInterface<ResolvedPublicTopic<never>['subscriber']>;
+  zoeTools: GuestInterface<ZoeTools>;
+};
```

Comment:
I had to stare at this for a while to see how they're the same and how they're different. I **suggest** we DRY this out:

```suggestion
export type PortfolioInstanceContext = {
  axelarChainsMap: AxelarChainsMap;
  chainHubTools: {
    getDenom: (brand: Brand) => Denom | undefined;
  };
  contractAddresses: EVMContractAddresses;
  inertSubscriber: GuestInterface<ResolvedPublicTopic<never>['subscriber']>;
  zoeTools: GuestInterface<ZoeTools>;
};

export type PortfolioBootstrapContext = PortfolioInstanceContext & {
  zoeTools: GuestInterface<ZoeTools>;
  makePortfolioKit: () => GuestInterface<PortfolioKit>;
};
```